### PR TITLE
Handle stale PCI address in mgmt port Init for DPU Host case

### PR DESCRIPTION
**Pull request by ykulazhenkov**
## 📑 Description
A DPU firmware settings change can cause the same physical
port to be re-enumerated under a different PCI address after
a host reboot. Previously, Init() only handled missing device
IDs (legacy annotations). Now it also detects when the
annotated device ID is no longer present in the allocator and
falls back to matching by PfId and FuncId.

## Additional Information for reviewers
Update `(mpdm *MgmtPortDeviceManager) Init()` logic to relax validation for the deviceId field

## ✅ Checks

- [ ] My code requires changes to the documentation
- [ ] if so, I have updated the documentation as required
- [ ] My code requires tests
- [ ] if so, I have added and/or updated the tests as required
- [ ] All the tests have passed in the CI 

## How to verify it
This fix addresses the following edge-case scenario:
1. Deploy ovn-kubernetes with nodes in "DPU host" mode and use `OVNKUBE_NODE_MGMT_PORT_DP_RESOURCE_NAME` to configure a management port.
2. Provision the DPU with the PCI_SWITCH_EMULATION=0 firmware setting.
3. Reboot the node.
4. ovn-kubernetes starts and runs correctly.
5. Re-provision the DPU with the PCI_SWITCH_EMULATION=1 firmware setting.
6. Reboot the node.
7. When the node starts, the PF/VF PCI addresses are different.
8. ovn-kubernetes cannot start because it has an outdated deviceId in the `k8s.ovn.org/node-mgmt-port` annotation 

The fix relaxes validation in `(mpdm *MgmtPortDeviceManager) Init()` to allow ovn-kubernetes to start even if the PF/VF PCI address has changed after a reboot.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved validation for management port devices to detect missing or stale device identifiers and ensure reservations reflect the live device pool.
  * Added a hardware-identifier fallback lookup to recover devices when IDs are absent; annotations are updated and an error is surfaced if no match is found.

* **Tests**
  * Added comprehensive unit tests covering initialization, recovery from stale/missing annotations, successful restoration, and failure paths.

### CODE REVIEW

**Comment by aserdean** [comment]
Thanks for the fix! The changes LGTM.

Would it be possible to add a unit test to capture this behavior?

CC: @npinaeva @trozet 

please also trigger the DPU CI for this PR

**Comment by ykulazhenkov** [comment]
> Would it be possible to add a unit test to capture this behavior?

Thanks, @aserdean, I added basic tests from these cases

File: go-controller/pkg/node/managementport/portDeviceManager_test.go [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+			DeferCleanup(wf.Shutdown)
+			mockDeviceDetails(sriovnetOpsMock, matchDevice, 3, 5)
+			kubeMock.On("SetAnnotationsOnNode", testNodeName,
+				mock.Anything).Return(nil).Once()
+
+			mpdm := NewMgmtPortDeviceManager(kubeMock, wf, testNodeName, allocator)
+			Expect(mpdm.Init()).NotTo(HaveOccurred())
+			Expect(mpdm.mgmtPortDetails["default"].DeviceId).
+				To(Equal(matchDevice))
+		})
```
[the remark this answers is not in the document]
Comment (ykulazhenkov) [reply go-controller/pkg/node/managementport/portDeviceManager_test.go:176]: I think the iteration in this case is deterministic since both the input and output configurations are lists.

**Comment by aserdean** [comment]
thank you for adding the testcase.


LGTM

CC: @npinaeva @trozet @tssurya
